internal/pkg/plugin/binary.go

```diff
@@ -72,65 +76,229 @@ func (m *Meta) Config() (*os.File, error) {
 	return os.Open(m.configName())
 }
 
-// NewFromImage returns a new meta object which hasn't yet been installed from
+// InstallFromSIF returns a new meta object which hasn't yet been installed from
 // a pointer to an on disk SIF. It will:
 //     1. Check that the SIF is a valid plugin
 //     2. Open the Manifest to retrieve name and calculate the path
 //     3. Copy the SIF into the plugin path
 //     4. Extract the binary object into the path
 //     5. Generate a default config file in the path
 //     6. Write the Meta struct onto disk in DirRoot
-func NewFromImage(fimg *sif.FileImage, libexecdir string) (*Meta, error) {
+func InstallFromSIF(fimg *sif.FileImage, libexecdir string) (*Meta, error) {
+	sylog.Debugf("Installing plugin from SIF to %q", libexecdir)
+
 	if !isPluginFile(fimg) {
-		return nil, fmt.Errorf("while opening sif file: not a valid plugin")
+		return nil, fmt.Errorf("while opening SIF file: not a valid plugin")
 	}
 
 	manifest := getManifest(fimg)
-	abspath, err := filepath.Abs(filepath.Join(libexecdir, pathFromManifest(manifest)))
+
+	plugindir := filepath.Join(libexecdir, DirRoot)
+
+	dstdir, err := filepath.Abs(filepath.Join(plugindir, pathFromName(manifest.Name)))
 	if err != nil {
 		return nil, fmt.Errorf("while getting absolute path to plugin installation: %s", err)
 	}
 
 	m := &Meta{
 		Name:    manifest.Name,
-		Path:    abspath,
+		Path:    dstdir,
 		Enabled: true,
 
 		fimg: fimg,
 	}
 
-	m.installTo(libexecdir)
-	return m, nil
+	err = m.install(plugindir)
+	return m, err
 }
 
-// installTo installs the plugin represented by m into libexecdir. This should
-// normally only be called in NewFromImage
-func (m *Meta) installTo(libexecdir string) {
+func Uninstall(name, libexecdir string) error {
+	sylog.Debugf("Uninstalling plugin %q from %q", name, libexecdir)
+
+	metas, err := List(libexecdir)
```

Comment:
Is it possible to change `List()` to return a `map[string]*Meta` to avoid having to iterate over a slice every time? I know it will likely never be an issue because of how quick Go is and the relatively few amount of plugins we expect to be installed